File: 2225-find-players-with-zero-or-one-losses/2225-find-players-with-zero-or-one-losses.cpp
class Solution {
public:
    vector<vector<int>> findWinners(vector<vector<int>>& matches) {
        vector<int> zeroLoses;
        vector<int> oneLose;
        unordered_map<int, int> map;

        for (const auto& team : matches) {
            map[team[0]] += 0;
            map[team[1]] += 1;
        }

        for (const auto& entry : map) {
            int i = entry.first;
            int loses = entry.second;

            if (loses == 0) {
                zeroLoses.push_back(i);
            } else if (loses == 1) {
                oneLose.push_back(i);
            }
        }

        sort(zeroLoses.begin(), zeroLoses.end());
        sort(oneLose.begin(), oneLose.end());

        return {zeroLoses, oneLose};
    }
};

File: 2225-find-players-with-zero-or-one-losses/2225-find-players-with-zero-or-one-losses.js
/**
 * @param {number[][]} matches
 * @return {number[][]}
 */
var findWinners = function(matches) {
    var zeroLoses = [];
    var oneLose = [];
    var map = new Map();

    for (const team of matches) {
        map.set(team[0], (map.get(team[0]) || 0) + 0);
        map.set(team[1], (map.get(team[1]) || 0) + 1);
    }

    for (const [i, loses] of map) {
        if (loses === 0) {
            zeroLoses.push(i);
        } else if (loses === 1) {
            oneLose.push(i);
        }
    }

    zeroLoses.sort((a, b) => a - b);
    oneLose.sort((a, b) => a - b);

    return [zeroLoses, oneLose];
};